### PR TITLE
USWDS - Accessibility: Added button functionality to usa-button links

### LINKED ISSUE #3690: Buttons: allow links that look like buttons to be triggered
USAGov accessibility [test results](https://docs.google.com/document/d/1g5beRCFgb5M7xYrEpBPuZy39AKYxtCXzT54FHTPiRgg/edit)

Visual Appearance of Control Does Not Match Function:
The Back button is coded as a link but is styled to look like a button. Visually sighted users see a button, but for screen reader users it is announced as a link. 
Add the ARIA role="button" so that screen readers announce it as a button. Additional JavaScript will also need to be added so that keyboard users can activate the button using both the Enter and Spacebar keys.

Button

## Todo
- [x] Allow links that look like buttons to be triggered by spacebar
- [ ] Update guidance on Buttons if needed.

### PR DESCRIPTION
**Pull request by mahoneycm**
# Summary

**Added spacebar activation to links styled as buttons**. This allows users to trigger these links the same way they would expect to trigger buttons. `usa-card` buttons have been replaced with links styled as buttons.

This PR has two main goals based on @thisisdano's comments in #3690:
- Links that are styled as buttons should be triggered by the spacebar without using `role="button"` [^1]
- `usa-card` buttons should be replaced with links [^2]

[^1]: https://github.com/uswds/uswds/issues/3690#issuecomment-951091169
[^2]: https://github.com/uswds/uswds/issues/3690#issuecomment-951095032

## Breaking change

:warning: This is a breaking change.

`usa-card` buttons have been replaced with links styled as buttons.

Before: 

```html
<div class="usa-card__footer">
    <button type="button" class="usa-button">Visit Florida Keys</button>
</div>
```

After:
```html
<div class="usa-card__footer">
    <a href="#" class="usa-button">Visit Florida Keys</a>
</div>
```

## Related issue

Closes #3690 

## Related PRs
[Changelog PR →](https://github.com/uswds/uswds-site/pull/2085)

## Preview link
[usa-card w/ links styled as buttons](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-button-group-a11y/iframe.html?args=&id=components-card--default&viewMode=story)

Preview link:

## Problem statement

Links styled as buttons should be triggered as such

## Solution

Create a simple JS file to target all anchors with the `usa-button` class and allow them to be triggered with the spacebar

Replace buttons in `usa-card` with `usa-button` anchor tags

## Testing and review

1. Visit `button-group` storybook preview
2. Turn on screen reader
3. Tab to "Back" link
4. Verify the screen reader reads it as a link
5. Press spacebar key
6. Verify that the link is followed
7. Visit usa-card
8. Verify that all buttons are now links with `usa-button` class
9. Verify spacebar & enter button trigger the links
   - It may help to update the links `href` to verify the page updates

Before opening this PR, make sure you’ve done whichever of these applies to you:
- [x] Confirm that this code follows the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [x] Run `git pull origin [base branch]` to pull in the most recent updates from your base and check for merge conflicts. (Often, the base branch is `develop`).
- [x] Run `npm run prettier:js` to format any JS updates.
- [x] Run `npm test` and confirm that all tests pass.

## Additional information

- Added `href="#"` to allow links to be targeted through tabbing. 
- Ran `npm run build:html` & `npm run build:storybook` for good measure.

### CODE REVIEW

File: src/components/11-button-groups/button-groups.njk [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <h6>Default</h6>
 <ul class="usa-button-group">
   <li class="usa-button-group__item">
-    <a href="#" class="usa-button usa-button--outline">Back</a>
+    <a href="#" role="button" class="usa-button usa-button--outline">Back</a>
```
Comment (mejiaj) [inline src/components/11-button-groups/button-groups.njk:4]: Please remove `role="button"`.

File: src/components/card/_card--basic.njk [hunk 2]
```diff
@@ -20,7 +20,7 @@
   </div>
   <div class="usa-card__footer">
     {%- block card_footer %}
-    <button class="usa-button">Visit Florida Keys</button>
+    <a href='#' class="usa-button">Visit Florida Keys</a>
```
Comment (mejiaj) [inline src/components/card/_card--basic.njk:23]: Double quotes on all `href` attributes.

**Review by mejiaj — CHANGES_REQUESTED** [review]
Thanks for updating this. Can you resolve conflicts and fix some formatting issues on `href`?

File: src/js/components/button.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const behavior = require('../utils/behavior');
+const keymap = require("receptor/keymap");
+const USA_BUTTON = `a[class*="usa-button"]`;
+
+const toggleUSABtn = function toggleUSABtn(event) {
```
Comment (mejiaj) [inline src/js/components/button.js:5]:
```suggestion
const toggleButton = (event) => {
```

File: src/js/components/button.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  event.preventDefault();
+  event.target.click();
+};
+
+const usaButton = behavior(
+  {
+    keydown: {
+      [USA_BUTTON]: keymap({
+        " ": toggleUSABtn
```
Comment (mejiaj) [inline src/js/components/button.js:14]:
```suggestion
        " ": toggleButton
```

**Review by mejiaj — CHANGES_REQUESTED** [review]
@mahoneycm thanks for all your work on this. 

Aside from the requested changes, would you mind creating a unit test for it under `spec/unit`?

**Comment by mahoneycm** [comment]
@thisisdano This was something I worked on a while back for accessibility. Remembered it was wondering if we should readdress?

**Comment by JackRyan1989** [comment]
Hi there! There's some interest from Login.gov in having this change implemented. Are there blockers or other considerations preventing this from being merged? Thanks

**Comment by mahoneycm** [comment]
@mejiaj Updated this PR for the latest USWDS! I also revised the PR description to make the work and problem more understandable. 

This included Dan's request to change `usa-card` buttons to links but that causes the markup to change. If we want to separate that task, we can remove it from this PR

**Review by mejiaj — APPROVED** [review]
LGTM, usually we require a unit test for every new script. I'm not sure on this one given the very limited scope.

**Review by amyleadem — CHANGES_REQUESTED** [review]
Looks good!
Confirmed that:
- links styled with `.usa-button` are activated by both `space` and `enter`
- links are identified as links in VoiceOver

For good measure, I also searched for any `<a>` tags with `type="button"` and couldn't find any. 

A couple small things:
- Could you update the PR summary to match the changelog/release notes format? (Some details outlined [here](https://github.com/uswds/uswds-site/blob/main/_data/changelogs/_CHANGELOG-README.md#writing-a-good-summary))
- Could you add details to the "Breaking changes" section so that users will know exactly what markup change they'll need to do?
- Also, this shouldn't need to block anything, but I'm wondering if we can make a story that explicitly demonstrates links styled as buttons. We have the example in the `button-group` story, but it is a bit hidden.

**Comment by mahoneycm** [comment]
@amyleadem Updated the PR description with those two changes!

I agree, I would like to highlight links styled as buttons somewhere as well. Do you think that this should be part of this PR or should I open up a new ticket for that work?

**Review by amyleadem — APPROVED** [review]
Thanks for updating the descriptions. I think it is fine to address the story as a follow-up. Will you open an issue so we don't lose track?

**Comment by mahoneycm** [comment]
Added changelog PR https://github.com/uswds/uswds-site/pull/2085

**Review by thisisdano — APPROVED** [review]

**Pull request merged**